### ISSUE TITLE
ICC Alpha last-minute clean up

### ISSUE THREAD

**Issue by robertocarroll** [robertocarroll/icc-alpha#104]
![icc-alpha](https://f.cloud.github.com/assets/1565953/2367557/0217d560-a76a-11e3-8959-866a0ca15b7f.jpg)

---
#1. Introduction

1.1 ~~Intro text~~
1.2 ~~Feedback form / questions / cheat sheet~~
Latest version of the feedback form is here:

https://docs.google.com/forms/d/1oZttxji5d3F38knR9vq5iQgk8q44GoD9_8xaqERxfLw/viewform

1.3 Blog posts 
## 1.4 ~~Presentation~~ 
#2. Home

http://sharepoint13vm.cloudapp.net/sites/icc/

2.1. Home layout 
1. ~~(RC) Create home layout. Here's the layout: https://dl.dropboxusercontent.com/u/6082092/03_homepage-D2-R3.pdf~~
2. ~~(MJ) Content for get involved, ASP, Resources~~
3. ~~(AT) Calendar mockup~~
4. ~~(SG) Implement home layout (1). Here's the content and markup for the top and the bottom sections: http://robertocarroll.com/icc-alpha/home.html~~

2.2. Menu
1. ~~(AT) Making the home layout have a different header. (SG) Remove breadcrumb from home header.~~ 
2. ~~(MJ) to confirm links for menu (shortcuts), top menu (shortcuts)~~

2.3. Map
1. (RC) ~~to work on new map version~~: http://robertocarroll.com/icc-alpha/map-new/

2.4. Footer links
1. ~~(RC) Make active links active and point to correct place.~~ 

---
#3. Situation - DRC

http://sharepoint13vm.cloudapp.net/sites/icc/drc
1. ~~(RC) - Ntaganda case summary - italics for French terms and February is in French - fevrier~~
2. ~~(RC) - Katanga case summary - italics for French terms, 2 dates with commas (remove) and remove the commas after Katanga.~~
3. Make the top video responsive 
4. ~~(MJ) Sent images and captions.~~ (SG)In the field section - include Outreach images as slideshow.
5. (SG) ~~Add `class="page-title"` to the title: `<h1 class="page-title">Democratic Republic of the Congo</h1>`~~
6. (SG) ~~Add link to situation intro: The Prosecutor [opened an investigation](http://www.icc-cpi.int/en_menus/icc/press%20and%20media/press%20releases/2004/Pages/the%20office%20of%20the%20prosecutor%20of%20the%20international%20criminal%20court%20opens%20its%20first%20investigation.aspx)~~
7. ~~Can we also add a large number for: 1 ONGOING APPEALS~~
8. (RC) Update case intros with charges (MJ email 12 Mar 2014, at 10:29) - Ngudjolo Chui Case and Ntaganda Case to do. 

---
#4. Case - Lubanga

http://sharepoint13vm.cloudapp.net/sites/icc/drc/lubanga

1.~~(SG) Remove numbers from people list~~
2. ~~(SG) View all change to: Who else is involved?~~
3. (AT) Check Armin adding calendar webpart
4. ~~Date style - 7 September 2013 not 07 September 2013 ~~
5. ~~(RC) Date format in timeline - 7 September 2013~~
6. (RC) ~~Legal Representative for Victims - capitalise~~ 
7. ~~(SG) Ask the Court - capitalise Court, eg on case layout title bottom~~
8. ~~(SG) "Appeals" - not "Appeal"~~
9.~~(SG) Fluid embed for timeline - not fixed width ~~
10. ~~(SG) Change in the field back to single image.~~
11.~~(SG) Update json for timeline~~

---
#5. Search
## 5.1 From top bar
1. Customise search - https://github.com/robertocarroll/icc-alpha/issues/87
   ## 5.2 From filter layout
   
   http://sharepoint13vm.cloudapp.net/sites/icc/court-records
2. Clean up mark up of filings layout https://github.com/robertocarroll/icc-alpha/issues/102

---
#6. Stages
1. ~~(AT) send latest case list and metadata.~~ (SG) to implement.  
2. ~~(SG) Tabs for the child case lists on each layout~~
3. (SG) Show top figure in an "active" state, ie I am on that layout. So it would not be clickable and the box would be in the hover state (blue backgroud / white text). Active class on stage which page is on or use Sharepoint-generated class specific to page? 
4. ~~(SG) Update intros with latest texts: https://gist.github.com/robertocarroll/9511819~~

---
## 6.1 Preliminary examinations

http://sharepoint13vm.cloudapp.net/sites/icc/preliminary-examinations
### 6.1.1 - Georgia

http://sharepoint13vm.cloudapp.net/sites/icc/georgia
1. ~~Remove time from list and format date DD MMMM YYYY~~
2. ~~Add style to this list~~
3. ~~Make the intro a decent line length (max-width 600px)~~
4. ~~Add Preliminary examination below title (same style as "Appeals" on Lubanga case~~ 

---
## 6.2 Situations under investigation

http://sharepoint13vm.cloudapp.net/sites/icc/situations
1. ~~DONE (RC) Lorem ipsum for situation intros - this is a question to highlight to Working Group - does PIDS have the freedom to write these.~~
2. ~~Check order of situations - reverse chronological~~ 

---
## 6.3 Pre-Trial

---
## 6.4 Trial

---
## 6.5 Appeals

---
## 6.6 Closed

---
#7. How the Court works

---
#8. Who's who
1. (RC) Static layouts for each. (MJ) to provide latest version of 
   
   8.1 Presidency
   8.2 Judges and Chambers
   8.3 Office of the Prosecutor
   8.4 Registry
   8.5 Defence
   8.6 Victims
   8.7 Witnesses
   8.8 Outreach

---
#9. Legal texts
## 9.1 Rome Statute
1. ~~(RC) HTML version - finish~~
2. (RC) Get example interactive article from FE (Chambers)
3. (SG) to incorporate into the main site.
## 9.2 Elements of Crimes
1. ~~(RC) HTML version - finish~~
2. (SG) to incorporate into the main site.

---
#10. Smaller screens / Mobile tweaks

~~1. Make the site title smaller~~ 
2. Make the nav fix only on bigger screens: https://gist.github.com/robertocarroll/9376400

---
#11. Calendar
1. ~~(AT) static example for home layout.~~
2.  ~~(AT) webpart for Lubanga case.~~

---
#12. Defendants

http://sharepoint13vm.cloudapp.net/sites/icc/defendants
1. Layout - fix mark up - http://robertocarroll.com/icc-alpha/defendants.html
2. ~~(SG) Lubanga not showing in list~~ 
3.  Bemba Kilolo Kabongo Babala Arido Case and Barasa Case  can appear under “Other” and “All” but no other categories
4. ~~(MG) New text for defendant descriptions.~~ 

---
#13. Get involved

~~Static content on home layout and in footer.~~ 

---
#14. Ask the Court

~~Static content on situations and case layouts~~

---
#15. ASP

~~Static content on home layout and in footer.~~

**Comment by simmigaba** [comment]
Hi Rob,
Further below tasks are completed. Pls strike-through in the text above after review:
1. (SG) Implement home layout (1)
   Here's the content and markup for the top and the bottom sections: http://robertocarroll.com/icc-alpha/home.html
   (SG) Remove breadcrumb from home header.
   .site-breadcrumb span.current {
   display: none;
   }
   2.4. Footer links
2. Make active links active and point to correct place.
3. (SG) Add class="page-title" to the title: <h1 class="page-title">Democratic Republic of the Congo</h1>
4. (SG) Add link to situation intro
   The Prosecutor opened an investigation 

6.1.1 - Georgia

http://sharepoint13vm.cloudapp.net/sites/icc/georgia

Remove time from list and format date DD MMMM YYYY
Add style to this list
Make the intro a decent line length (max-width 600px)
Add Preliminary examination below title (same style as "Appeals" on Lubanga case

**Comment by simmigaba** [comment]
Hi Rob,
Further below list of tasks are completed. Pls strike-through in the text above after review:
1. (RC) to work on new map version: http://robertocarroll.com/icc-alpha/map-new/
(SG) In the field section - include Outreach images as slideshow.
7. Can we also add a large number for: 1 ONGOING APPEALS
Clean up mark up of filings layout #102
(SG) Update intros with latest texts: https://gist.github.com/robertocarroll/9511819
Check order of situations - reverse chronological
(SG) to incorporate into the main site. (Rome Statue)
(SG) to incorporate into the main site. (Elements of Crime)
Layout - fix mark up - http://robertocarroll.com/icc-alpha/defendants.html

On one hand I am happy to complete many changes/ enhancements this week, on the other hand I have to mention that changes related to Section  6. Stages are not quick fixes and require significant amount of coding and testing effort. Let me explain further for each point:
1. (SG) Tabs for the child case lists on each layout
-- Currently we use button controls in ASPX e.g.
 asp:Button ID ="TrialC" text="Cases at the Trial commenced stage" OnClick="Trial_Commenced" runat="server"/
This button id is registered as a Trigger for Ajax Update Panel, so when user clicks on this button it triggers refresh of the Panel asynchronously. And content to display is generated by button event of OnClick.
  asp:UpdatePanel ID="UpdatePanelTial" runat="server"
            ContentTemplate
               asp:Table Width="90%" CellPadding="20" ID="CaseStageListTableInitial" runat="server">/asp:Table
               asp:Table Width="90%" CellPadding="20" ID="CaseStageListTable" runat="server">/asp:Table
            /ContentTemplate
   Triggers>
        asp:AsyncPostBackTrigger ControlID="TrialC" /
   /Triggers>
 /asp:UpdatePanel

Now if I want to replace buttons with HTML tab then I have to generate this HTML controls in C# code and then have to change event handling etc. Also in past I had issues with mix up of controls in HTML and using Ajax Update panel. Therefore I need to analyze and test it more. Only quick win will be if you want to propose any style/css for ASP buttons so that they have look and feel like HTML tabs. 
1. (SG) Metadata to implement.
   -- For this earlier we only had one field in Metadata taxonomy to represent phase of case. As now its two level, so I have created new parent and child fields for phase in Taxonomy and have also populated values for all cases as per XLS given by Armin. This part was easy to do. But now as old meta data field value was used in multiple web parts on many pages and was also used for deriving counts, deriving filter logic etc, so I have the code to change at many places. And more then that it involves regression testing effort. Hence I want to change the code only after today's presentation as otherwise it will have potential to break many things in last hour.
2. (SG) Show top figure in an "active" state, ie I am on that layout. -- Again for this the things are little bit mixed up between c#, ASP and CSS styling. And I am using only one common webpart for all Stage pages where content is created dynamically based on stage filter value configured in webpart properties. Again this is something which can be done, but is suited to do after today's presentation.

I believe you will share my views in this case, because as you know we do not have a separate dev, test and demo environment. So for me its not easy to develop and test repeatedly while users are reviewing or a demo is planned. Thanks.

**Comment by simmigaba** [comment]
I have now updated the Stage pages as per the latest meta data values for Case Phases. Though I remember that in one of the requirements you also ask additional buttons for Pre-trial or Appeals Stage page. Can you please confirm this requirement again along with the details.

**Comment by robertocarroll** [comment]
The second sheet of Cases-stages.xlsx shows child categories for Pre-Trial and Trial. 

The buttons on each stage layout need to reflect the metadata: 

Pre-Trial  
        Before suspect’s appearance  
    After suspect’s appearance  
    Charges not confirmed
Trial  
        In preparation

##     Ongoing

Appeals only seems to have one child - Ongoing - so it doesn't make sense to have a button. 

So in summary - you need to remove the trial closed button from the trial stage and add three buttons to the pre-trial stage. 

Let me know if that it OK.

**Comment by simmigaba** [comment]
The buttons on Pre-trial and Trial pages are now implemented to reflect the metadata. Also below item is completed:
(SG) Show top figure in an "active" state, ie I am on that layout. So it would not be clickable and the box would be in the hover state (blue backgroud / white text). Active class on stage which page is on or use Sharepoint-generated class specific to page?
(SG) to implement.

Please review.

**Comment by simmigaba** [comment]
As discussed yesterday
1. "States Parties to the ICC Statute" and "Twelth Session" sections have been moved between Stages and Resource Library sections of the page
1. For DRC page, the text below the numbers seem to be appearing fine in all browsers on my side (tested with chrome, firefox and internet explorer). attached screenshot (DRC-textsize.jpg) for reference. Kindly let me know in which browser you are getting the cramped text.
   ![drc-textsize](https://f.cloud.github.com/assets/5706403/2457040/fbbe054a-af2a-11e3-8998-6dbaed7b24f4.jpg)
2. Tab title for DRC page has been set correctly, also earlier all pages were showing tab titles as ICC<<Page Title>>, I have now changed it to show only <<PageTitle>>
   Here are the steps to edit page title (if you need to do it in future),go to the specific page, under the ribbon go to Page tab, click Edit Properties, set Title property. Check in the page to make your changes visible to rest of the users.

**Comment by robertocarroll** [comment]
Closing this issue. Well done! Thanks to all for the great work.
